modules/ldap-auto-syncing.adoc

```diff
@@ -0,0 +1,246 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: ldap-group-syncer
+  labels:
+    app: cronjob-ldap-group-sync
+rules:
+  - apiGroups:
+      - ''
+      - user.openshift.io
+    resources:
+      - groups
+    verbs:
+      - get
+      - list
+      - create
+      - update
+----
++
+// TODO: any of these fields need pointed out?
+
+. Create the cluster role:
++
+[source,terminal]
+----
+$ oc create -f ldap-sync-cluster-role.yaml
+----
+
+. Define a cluster role binding to bind the cluster role to the service account:
++
+.Example `ldap-sync-cluster-role-binding.yaml`
+[source,yaml]
+----
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ldap-group-syncer
+  labels:
+    app: cronjob-ldap-group-sync
+subjects:
+  - kind: ServiceAccount
+    name: ldap-group-syncer <1>
+    namespace: ldap-sync
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: ldap-group-syncer <2>
+----
+<1> Reference to the service account created earlier in this procedure.
+<2> Reference to the cluster role created earlier in this procedure.
+
+. Create the cluster role binding:
++
+[source,terminal]
+----
+$ oc create -f ldap-sync-cluster-role-binding.yaml
+----
+
+. Define a config map that specifies the LDAP config:
++
+.Example `ldap-sync-config-map.yaml`
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: ldap-group-syncer
+  namespace: ldap-sync
+  labels:
+    app: cronjob-ldap-group-sync
+data:
+  ldap-group-sync.yaml: | <1>
+    kind: LDAPSyncConfig
+    apiVersion: v1
+    url: ldap://10.0.0.0:389 <2>
+    insecure: false
+    bindDN: cn=admin,dc=example,dc=com <3>
+    bindPassword:
+      file: "/etc/secrets/bindPassword"
+    ca: /etc/ldap-ca/ca.crt
+    rfc2307:
+        groupsQuery:
+            baseDN: "ou=groups,dc=example,dc=com" <4>
+            scope: sub
+            filter: "(objectClass=groupOfMembers)"
+            derefAliases: never
+            pageSize: 0
+        groupUIDAttribute: dn
+        groupNameAttributes: [ cn ]
+        groupMembershipAttributes: [ member ]
+        usersQuery:
+            baseDN: "ou=users,dc=example,dc=com" <5>
+            scope: sub
+            derefAliases: never
+            pageSize: 0
+        userUIDAttribute: dn
+        userNameAttributes: [ uid ]
+        tolerateMemberNotFoundErrors: false
+        tolerateMemberOutOfScopeErrors: false
+----
+<1> Define the desired LDAP config. This example uses the RFC2307 schema; adjust values as necessary.
+<2> Edit the url as required.
+<3> Edit the `bindDN` as required.
+<4> Edit the groupsQuery `baseDN` as required.
+<5> Edit the usersQuery `baseDN` as required.
++
+// TODO: point out additional fields as necessary. Also work on better callouts.
+
+. Create the config map:
++
+[source,terminal]
+----
+$ oc create -f ldap-sync-config-map.yaml
+----
+
+. Define a cron job:
++
+.Example `ldap-sync-cron-job.yaml`
+[source,yaml]
+----
+kind: CronJob
```

Comment:
@soltysh I'll ask some specific questions on some fields, but if you could take a look at the overall cron job config here and let me know if anything looks like it isn't needed or should be something else, please let me know! 

Or if anything needs a callout as something that they probably will/should change.